activesupport/lib/active_support/values/time_zone.rb

```diff
@@ -269,10 +269,7 @@ def load_country_zones(code)
         end
 
         def zones_map
-          @zones_map ||= begin
-            MAPPING.each_key { |place| self[place] } # load all the zones
-            @lazy_zones_map
-          end
+          @zones_map ||= Hash[MAPPING.keys.map { |place| [place, self[place]] }]
```

Comment:
This can be written as:

``` ruby
@zones_map ||= MAPPING.transform_values { |tz| self[tz] }
```

Reply:
Not quite. It seems important to use the _keys_ of `MAPPING` to perform lookups using `#[]` because the argument to `#[]` ends up on the `name` attribute of the returned `ActiveSupport::TimeZone` instance.

```ruby
> Hash[MAPPING.keys.map { |k| [k, self[k]] }]['UTC'].name
=> "UTC"
> MAPPING.transform_values { |v| self[v] }['UTC'].name
=> "Etc/UTC"
```

If there's a nice way to use `Hash#transform_values` in a way that I have access to the key, it would work, but I don't think that exists.

```
> MAPPING.transform_values { |_, k| self[k] }['UTC'].name
=> "UTC"
```

Reply:
Sorry about that - you can use `each_with_object` to get both the key and the value, e.g.

``` ruby
@zones_map ||= MAPPING.each_with_object({}) { |(name, _), zones| zones[name] = self[name] }
```

Reply:
Done. 69b96af